app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -0,0 +1,365 @@
+    serverTextView = (TextView) findViewById(R.id.server_textView);
+
+    if (savedInstanceState != null) {
+      serverTextView.setText(
+          getString(R.string.server_started_message) + " " + savedInstanceState.getString(
+              IP_STATE_KEY));
+      startServerButton.setText(getString(R.string.stop_server_label));
+    }
+
+    FragmentManager fragmentManager = getSupportFragmentManager();
+    FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
+    ZimFileSelectFragment fragment = new ZimFileSelectFragment();
+    fragmentTransaction.add(R.id.frameLayoutServer, fragment);
+    fragmentTransaction.commit();
+
+    serviceIntent = new Intent(this, HotspotService.class);
+    Context context = this;
+
+    startServerButton.setOnClickListener(new View.OnClickListener() {
+      @Override public void onClick(View v) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+          toggleHotspot();
+        } else {
+          //TO DO: show Dialog() + within that add check mobile Data check later.
+          //if (isMobileDataEnabled(context)) {
+          //  mobileDataDialog();
+          //} else {
+          if (flag) {
+            serverStopped();
+            isStarted = false;
+          } else {
+            startHotspotDialog();
+          }
+          //}
+        }
+      }
+    });
+  }
+
+  @RequiresApi(api = Build.VERSION_CODES.O)
+  void toggleHotspot() {
+    //Check if location permissions are granted
+    if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION)
+        == PackageManager.PERMISSION_GRANTED) {
+      if (checkHotspotState(this)) //If hotspot is already enabled, turn it off
+      {
+        startService(ACTION_TURN_OFF_AFTER_O);
+      } else //If hotspot is not already enabled, then turn it on.
+      {
+        setupLocationServices();
+      }
+    } else {
+      //Ask location permission if not granted
+      ActivityCompat.requestPermissions(this,
+          new String[] { Manifest.permission.ACCESS_FINE_LOCATION },
+          MY_PERMISSIONS_ACCESS_FINE_LOCATION);
+    }
+  }
+
+  // This method checks if mobile data is enabled in user's device.
+  static boolean isMobileDataEnabled(Context context) {
+    boolean enabled = false;
+    ConnectivityManager cm =
+        (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+    try {
+      Class cmClass = Class.forName(cm.getClass().getName());
+      Method method = cmClass.getDeclaredMethod("getMobileDataEnabled");
+      method.setAccessible(true);
+      enabled = (Boolean) method.invoke(cm);
+    } catch (Exception e) {
+      Log.e("TAG", e.toString());
+    }
+    return enabled;
+  }
+
+  //This method sends the user to data usage summary settings activity
+  private void disableMobileData() {
+    Intent intent = new Intent();
+    intent.setComponent(new ComponentName("com.android.settings",
```

Comment:
`startActivity(new Intent(android.provider.Settings.ACTION_WIRELESS_SETTINGS));`
Maybe this would function equivalently? addressing a component by its name as a raw string makes me anxious

Reply:
Fixed. 63abcae